### ISSUE TITLE
server crash: server watchdog detected that a single server tick took 60 seconds.

### ISSUE THREAD

**Issue by supremedutchgamer** [The-Nexus-Project/Limitless-3#71]
[crash-2021-01-01_02.11.50-server.txt](https://github.com/The-Nexus-Project/Limitless-3/files/5758573/crash-2021-01-01_02.11.50-server.txt)

started up a fresh server, around 20 to 30 minutes in, the entire server froze up and just stopped responding. what was being done at the time: one person was exploring, another was building a door using tinkers mechworks. 

affected version: 1.8.2, on a server.

**Comment by Irgendwer01** [comment]
@supremedutchgamer Temporary workaround: set max-tick-time to -1 in server.properties

**Comment by Gbergz** [comment]
The default server.properties has that set to 120000, meaning you don't have the official server files.
But yes, try disabling by setting it to "-1".

**Comment by Irgendwer01** [comment]
@Gbergz I have the Official Server files and not modified them very heavy, i just recommended to set the maxticktime to -1 as temporary workaround since the server from him crashes due to Ticks

**Comment by supremedutchgamer** [comment]
> The default server.properties has that set to 120000, meaning you don't have the official server files.
> But yes, try disabling by setting it to "-1".

this resolves the server shutting down, but does not stop it from infinitely hanging on a single tick. which was the actual issue. the game isnt supposed to take 60 seconds for a single tick.

**Comment by Irgendwer01** [comment]
@supremedutchgamer do u have actually tested if it works?

**Comment by supremedutchgamer** [comment]
> @supremedutchgamer do u have actually tested if it works?

first of all, yes i have.
second of all. i am not an idiot. 

the max ticktime setting is there to prevent servers from infinitely hanging. it achieves this by stopping the server if a single tick takes more than the set amount of time. this can be paired with a simple loop function in your start.bat file to make the server auto-restart if it hangs for a minute. 

normally, the game runs at 20 ticks per second, meaning every tick is supposed to take 50ms. if a single tick ends up taking 60 seconds, that is an issue. regardless of wether the server shuts itsself down at that point or not. by increasing the maximum tick time, you dont actually fix the issue, just how long it takes for the server to actually do something about hanging. setting it to -1 disables the function, simply allowing the server to hang indefinitely. 

the maximum tick time setting on servers has nothing to do with why the server is hanging. it is merely a tool for preventing the server from hanging indefinitely. so increasing the time it takes for that to kick in does not have any effect on wether or not the server will hang. merely for how long itll hang before the server shuts itsself off. 

even if we were to assume that after those 60 seconds, it'd finish processing the tick, in which case increasing the max tick time would stop the server from shutting down. that'd still mean that a fresh server is running at 1TPM. while the game is barely playable at 10TPS. 

in short:
does setting max-tick-time to -1 prevent the server from "crashing" (forced shutdown)? yes.
does setting max-tick-time to -1 prevent the server from infinitely hanging on one tick? no.
what was my issue? the server infinitely hanging on one tick.
is my issue resolved? no it is not.

**Comment by Gbergz** [comment]
This has hopefully been fixed in the next pack update. Looks like it was an issue with Ice and Fire & world deadlocks.
